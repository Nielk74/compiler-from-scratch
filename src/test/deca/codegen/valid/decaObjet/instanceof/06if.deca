// Description:
//    instanceof dans un if avec comparaison avec un boolean
//
// Resultats:
//    a instanceof A
//    a not instanceof B
//
// Historique:
//    cree le 17/06/2022

class A {}

class B {}

{
    A a = new A();
    boolean b = true;

    if (b == a instanceof A) {
        println("a instanceof A");
    } else {
        println("a not instanceof A");
    }

    if (b == a instanceof B) {
        println("a instanceof B");
    } else {
        println("a not instanceof B");
    }
}