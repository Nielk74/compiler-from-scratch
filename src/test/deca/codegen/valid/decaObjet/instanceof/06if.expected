a instanceof A
a not instanceof B
